src/Database/Driver/Sqlserver.php

```diff
@@ -187,12 +187,15 @@ public function enabled(): bool
      * Prepares a sql statement to be executed
      *
      * @param string|\Cake\Database\Query $query The query to prepare.
+     * @param array $options The driver options to set on the statement.
      * @return \Cake\Database\StatementInterface
      */
-    public function prepare($query): StatementInterface
+    public function prepare($query, array $options = []): StatementInterface
     {
         $this->connect();
-        $options = [PDO::ATTR_CURSOR => PDO::CURSOR_SCROLL];
+        if (!isset($options[PDO::ATTR_CURSOR])) {
+            $options = [PDO::ATTR_CURSOR => PDO::CURSOR_SCROLL] + $options;
```

Comment:
```suggestion
            $options[PDO::ATTR_CURSOR] = PDO::CURSOR_SCROLL;
```

Reply:
Unfortunately this would throw the same exception as: https://github.com/cakephp/cakephp/pull/14814#discussion_r457783780

Reply:
That's pretty silly that the options need to be set in a specific order.

Reply:
I know right, seems they do a validation per option set, instead of the whole array.